safebrowser.go

```diff
@@ -405,30 +409,34 @@ func (sb *SafeBrowser) LookupURLs(urls []string) (threats [][]URLThreat, err err
 	// TODO: There are some optimizations to be made here:
 	//	1.) We could force a database update if it is in error.
 	//  However, we must ensure that we perform some form of rate-limiting.
-	//	2.) We should batch all of the partial hashes together such that we
-	//	call api.HashLookup only once.
+	hashes := make(map[hashPrefix]string)
```

Comment:
I don't follow on this either. Yes, this hashes is hash2pattern. but hash2idx store the index of "threats [][]URLThreat".